src/generators/java/JavaGenerator.ts

```diff
@@ -48,4 +50,46 @@ export class JavaGenerator extends AbstractGenerator<JavaOptions> {
     const result = await renderer.runSelfPreset();
     return RenderOutput.toRenderOutput({result, dependencies: renderer.dependencies});
   }
+
+  /**
+   * Generates all the models to an output directory. 
+   * 
+   * This function is invasive, as it overwrite any existing files with the same name as the model.
+   * 
+   * @param input
+   * @param outputDirectory where you want the models generated to
+   * @param packageName for the models
+   */
+  public async generateToFile(input: Record<string, unknown> | CommonInputModel, outputDirectory: string, packageName: string): Promise<OutputModel[]> {
+    const generatedModels = await this.generateFullOutput(input, packageName);
+    for (const outputModel of generatedModels) {
+      const outputFilePath = path.resolve(outputDirectory, `${FormatHelpers.toPascalCase(outputModel.model.$id || 'undefined')}.java`);
+      await FileHelpers.writeToFile(outputModel.result, outputFilePath);
+    }
+    return Promise.resolve(generatedModels);
+  }
+
+  /**
+   * Generates the full output of a model, instead of a scattered model.
+   * 
+   * OutputModels result is no longer the model itself, but including package, package dependencies and model dependencies.
+   * 
+   * @param input 
+   * @param packageName for the models
+   */
+  public async generateFullOutput(input: Record<string, unknown> | CommonInputModel, packageName: string): Promise<OutputModel[]> {
+    if (isReservedJavaKeyword(packageName)) {
+      return Promise.reject(`You cannot use reserved Java keyword (${packageName}) as package name, please use another.`);
+    }
+
+    const generatedModels = await this.generate(input);
+    return Promise.resolve(generatedModels.map((outputModel) => {
+      const modelDependencies = outputModel.model.getNearestDependencies().map((dependencyModel) => { return `import ${packageName}.${FormatHelpers.toPascalCase(dependencyModel || 'undefined')};`;});
```

Comment:
The `undefined` fallback can produces something like:

```java
import somePackage.Undefined
```

and I don't think so that it's good solution. Please give me one case where such a situation will be possible. If not, please filter undefined models.